metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/lineage.py

```diff
@@ -195,6 +196,128 @@ def compute_bigquery_lineage_via_exported_bigquery_audit_metadata(
             )
             raise e
 
+    def compute_bigquery_lineage_via_catalog_lineage_api(
+        self, project_id: str
+    ) -> Dict[str, Set[str]]:
+        """
+        Uses Data Catalog API to request lineage metadata. Please take a look at the API documentation for more details.
+
+        NOTE: It's necessary for you to enable the service API in your Google Cloud Project.
+
+        Args:
+            project_id(str): Google project id. Used to search for tables and datasets.
+
+        Returns:
+            Dict[str, Set[str]] - A dictionary, where keys are the downstream table's identifier and values is a set
+            of upstream tables identifiers.
+        """
+        logger.info("Populating lineage info via Catalog Data Linage API")
+        #  Data Catalog API version 0.2.0 don't export views lineage, but future versions can cover it
+        #  NOTE: Users can enable to parse views DDL to build views lineage
+        if self.config.include_views:
+            logger.warning(
+                "It's not possible to extract views lineage from Catalog API. Views will be ignored."
+            )
+
+        # Regions to search for BigQuery tables: projects/{project_id}/locations/{region}
+        enabled_regions: List[str] = ["US", "EU"]
+
+        try:
+            lineage_client: lineage_v1.LineageClient = lineage_v1.LineageClient()
+            bigquery_client: BigQueryClient = BigQueryClient()
+            # Filtering datasets
+            datasets = list(
+                filter(
+                    lambda x: self.config.dataset_pattern.allowed(x.dataset_id),
+                    list(bigquery_client.list_datasets(project_id)),
+                )
+            )
+            project_tables = []
+            for dataset in datasets:
+                # Enables only tables where type is TABLE (removes VIEWS)
+                project_tables.extend(
+                    [
+                        table
+                        for table in bigquery_client.list_tables(dataset.dataset_id)
+                        if table.table_type == "TABLE"
+                    ]
+                )
+
+            # Convert project tables to <project_id>.<dataset_id>.<table_id> format
+            project_tables = list(
+                map(
+                    lambda table: "{}.{}.{}".format(
+                        table.project, table.dataset_id, table.table_id
+                    ),
+                    project_tables,
+                )
+            )
+
+            # Filtering tables
+            if self.config.table_pattern:
```

Comment:
Hmm, understand. So, it's better to not apply filtering, right?

Reply:
yes, exactly as if you use apply filter it can happen you create phantom nodes while if you only set edges between existing datasets this can't happen